src/api/auth/Dockerfile

```diff
@@ -1,12 +1,50 @@
-FROM node:lts-alpine as base
-RUN apk add dumb-init
-WORKDIR /app
-COPY --chown=node:node ./package.json ./
+## Base ########################################################################
+# Use a larger node image to do the build for native deps (e.g., gcc, pytyhon)
+FROM node:lts as base
 
-FROM base AS dependencies
-RUN npm install --only=production --no-package-lock
+# Reduce npm log spam and colour during install within Docker
+ENV NPM_CONFIG_LOGLEVEL=warn
+ENV NPM_CONFIG_COLOR=false
 
-FROM dependencies as deploy
-COPY --chown=node:node ./src ./src
+# We'll run the app as the `node` user, so put it in their home directory
+WORKDIR /home/node/app
+# Copy the package.json and lock file over
+COPY package*.json /home/node/app/
+
+## Development #################################################################
+# Define a development target that installs devDeps and runs in dev mode
+FROM base as development
+WORKDIR /home/node/app
+# Install (not ci) with dependencies, and for Linux vs. Linux Musl (which we use for -alpine)
+RUN npm install
+# Copy the source code over
+COPY --chown=node:node . /home/node/app/
+# Switch to the node user vs. root
+USER node
+# Start the app in debug mode so we can attach the debugger
+CMD ["npm", "run", "dev"]
```

Comment:
how can I test this?

Reply:
When you do a `build`, you specify a `--target <stage name>` and it will build that stage only:

```sh
docker build --target development -t auth:dev .
```

See https://docs.docker.com/develop/develop-images/multistage-build/#stop-at-a-specific-build-stage

Reply:
In a development `docker-compose.yml`, you'd use it like this:

```yml
services:
  auth:
    # Override the service so it builds the Dockerfile in development mode
    build:
      context: src/api/auth
      dockerfile: Dockerfile
      target: development
```